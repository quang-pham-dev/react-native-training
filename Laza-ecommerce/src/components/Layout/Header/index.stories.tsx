import React from 'react';
import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react-native';
import CenterView from '../../../../storybook/stories/CenterView';
import Header from './index';

storiesOf('Layout Header', module)
  .addDecorator(getStory => <CenterView>{getStory()}</CenterView>)
  .add('Default', () => (
    <Header
      navigation={{
        dispatch: action('clicked')
      }}
    />
  ));
