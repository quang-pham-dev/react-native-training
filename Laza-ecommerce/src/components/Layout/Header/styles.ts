import { StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Metrics from 'themes/Metrics';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '100%'
  },
  menuWrapper: {
    backgroundColor: Colors.primaryBackground
  },
  menuIcon: {
    width: Metrics.icons.semiLarge,
    height: Metrics.icons.semiLarge
  },

  cartWrapper: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  cartIcon: {
    width: Metrics.icons.semiLarge,
    height: Metrics.icons.semiLarge
  }
});

export default styles;
