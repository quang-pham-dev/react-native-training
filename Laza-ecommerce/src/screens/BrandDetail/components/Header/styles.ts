import { Platform, StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Metrics from 'themes/Metrics';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '100%',
    marginTop: Platform.OS === 'android' ? 40 : 45,
    position: 'absolute',
    backgroundColor: Colors.primaryBackground
  },

  brandLogoWrapper: {
    height: 45,
    width: 70,
    padding: Metrics.padding.sm,
    borderRadius: 10,
    backgroundColor: Colors.secondaryBackground
  },

  brandLogo: {
    width: '100%',
    height: '100%',
    resizeMode: 'contain'
  }
});

export default styles;
