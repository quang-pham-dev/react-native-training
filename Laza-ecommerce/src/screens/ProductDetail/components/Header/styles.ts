import { Platform, StyleSheet } from 'react-native';

// Themes
import Colors from 'themes/Colors';
import Fonts from 'themes/Fonts';
import Metrics from 'themes/Metrics';
import { height } from 'themes/Dimensions';

const styles = StyleSheet.create({
  headerContainer: {
    height: height * 0.5,
    backgroundColor: Colors.lightGray,
    alignItems: 'center'
  },

  actionsWrapper: {
    position: 'absolute',
    width: '100%',
    marginTop: Platform.OS === 'android' ? 40 : 45,
    zIndex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: Metrics.padding.lg
  },

  headerImageWrapper: {
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center'
  },

  image: {
    width: '100%',
    height: '100%',
    resizeMode: 'cover'
  }
});

export default styles;
