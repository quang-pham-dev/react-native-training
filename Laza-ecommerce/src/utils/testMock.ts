export const navigationMock: any = {
  addListener: jest.fn(),
  dangerouslyGetParent: jest.fn(),
  dismiss: jest.fn(),
  dispatch: jest.fn(),
  emit: jest.fn(),
  getParam: jest.fn(),
  goBack: jest.fn(),
  isFirstRouteInParent: jest.fn(),
  isFocused: jest.fn(),
  navigate: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  push: jest.fn(),
  replace: jest.fn(),
  reset: jest.fn(),
  router: undefined,
  setParams: jest.fn(),
  route: undefined,
  state: {
    key: '',
    routeName: '',
    params: {},
    path: '',
    index: 1,
    isTransitioning: false,
    routes: []
  }
};
