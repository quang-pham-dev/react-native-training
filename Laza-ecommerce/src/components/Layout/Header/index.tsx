import React, { memo, useCallback } from 'react';
import { Image, Pressable, View } from 'react-native';

// LIBS
import { DrawerActions } from '@react-navigation/native';

// Types
import { IHeaderProps } from 'types/screens/Layout';

// Themes
import IMAGES from 'themes/Images';

// Styles
import styles from './styles';

const Header = ({ navigation }: IHeaderProps) => {
  // Toggle menu open
  const handleToggleMenu = useCallback(() => {
    navigation.dispatch(DrawerActions.openDrawer());
  }, []);

  return (
    <View style={styles.container}>
      <Pressable style={styles.menuWrapper} onPress={handleToggleMenu}>
        <Image style={styles.menuIcon} source={IMAGES.iconMenu} />
      </Pressable>
      <Pressable style={styles.cartWrapper}>
        <Image style={styles.cartIcon} source={IMAGES.iconCart} />
      </Pressable>
    </View>
  );
};

export default memo(Header);
