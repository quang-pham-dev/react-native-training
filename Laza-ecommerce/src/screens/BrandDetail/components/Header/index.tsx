import React, { memo, useCallback } from 'react';
import { Image, ImageBackground, Pressable, View } from 'react-native';

// LIBS
import isEqual from 'react-fast-compare';

// Components
import Button from 'components/Button';

// Theme
import IMAGES from 'themes/Images';

// Style
import styles from './styles';

type HeaderProps = {
  navigation: any;
  logoUrl: any;
};

const Header = ({ logoUrl, navigation }: HeaderProps) => {
  // handle back button
  const handlePressBackIcon = useCallback(() => {
    navigation.goBack();
  }, []);

  return (
    <View style={styles.container}>
      <Button icon={IMAGES.iconBack} onPress={handlePressBackIcon} type='Circle' />
      <View style={styles.brandLogoWrapper}>
        {Boolean(logoUrl) && <Image style={styles.brandLogo} source={{ uri: logoUrl }} />}
      </View>
      <Button icon={IMAGES.iconCart} type='Circle' />
    </View>
  );
};

export default memo(Header, isEqual);
