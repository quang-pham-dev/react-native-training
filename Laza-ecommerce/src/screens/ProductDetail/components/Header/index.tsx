import React, { useCallback } from 'react';
import { ImageBackground, View } from 'react-native';

// Components
import Button from 'components/Button';

// Theme
import IMAGES from 'themes/Images';

// Types
import { IProductDetailHeaderProps } from 'types/screens/ProductDetail';

// Style
import styles from './styles';

const Header = ({ source, navigation }: IProductDetailHeaderProps) => {
  // handle back button
  const handlePressBack = useCallback(() => {
    navigation.goBack();
  }, []);

  return (
    <View style={styles.headerContainer}>
      <View style={styles.actionsWrapper}>
        <Button icon={IMAGES.iconBack} onPress={handlePressBack} type='Circle' />
        <Button icon={IMAGES.iconCart} type='Circle' />
      </View>
      {/* End action header */}

      <View style={styles.headerImageWrapper}>
        {Boolean(source) && <ImageBackground style={styles.image} source={{ uri: source }} />}
      </View>
      {/* End Header block */}
    </View>
  );
};

export default Header;
