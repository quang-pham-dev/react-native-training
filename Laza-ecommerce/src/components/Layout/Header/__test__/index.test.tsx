import React from 'react';
import { Pressable } from 'react-native';

// LIBS
import renderer from 'react-test-renderer';

// Components
import Header from 'components/Layout/Header';

// Utils
import { navigationMock } from 'utils/testMock';

describe('Header Component', () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
      toggleDrawer: jest.fn(),
      dispatch: jest.fn(),
    },
  };

  const tree = renderer.create(<Header {...props} />);

  test('should render correctly', () => {
    const component = tree.toJSON();
    expect(component).toMatchSnapshot();
  });

  test('should call function handleToggleMenu', () => {
    const Pressabled = tree.root.findAllByType(Pressable)[0];
    Pressabled.props.onPress();
    expect(props.navigation.dispatch).toHaveBeenCalled();
  });
});
