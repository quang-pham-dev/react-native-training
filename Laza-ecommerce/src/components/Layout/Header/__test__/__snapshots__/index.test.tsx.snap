// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 45,
          "width": 45,
        }
      }
    />
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 45,
          "width": 45,
        }
      }
    />
  </View>
</View>
`;
